plugins/gcp/resoto_plugin_gcp/resources/billing.py

```diff
@@ -100,23 +100,38 @@ class GcpService(GcpResource):
         "id": S("serviceId"),
         "tags": S("labels", default={}),
         "name": S("name"),
+        "display_name": S("displayName"),
         "ctime": S("creationTimestamp"),
         "description": S("description"),
         "link": S("selfLink"),
         "label_fingerprint": S("labelFingerprint"),
         "deprecation_status": S("deprecated", default={}) >> Bend(GcpDeprecationStatus.mapping),
         "business_entity_name": S("businessEntityName"),
-        "display_name": S("displayName"),
-        "service_id": S("serviceId"),
     }
 
     business_entity_name: Optional[str] = field(default=None)
     display_name: Optional[str] = field(default=None)
-    service_id: Optional[str] = field(default=None)
 
-    def post_process(self, graph_builder: GraphBuilder, source: Json) -> None:
-        for sku in GcpSku.collect_resources(graph_builder, parent=self.id):
-            graph_builder.add_edge(self, node=sku)
+    @classmethod
+    def collect(cls: Type[GcpResource], raw: List[Json], builder: GraphBuilder) -> List[GcpResource]:
+        # Additional behavior: iterate over list of collected GcpService and for each:
+        # - collect related GcpSku
+        result: List[GcpResource] = super().collect(raw, builder)  # type: ignore
+        SERVICES_COLLECT_LIST = [
+            "Compute Engine",
+        ]
+        service_names = [
+            service.name for service in cast(List[GcpService], result) if service.display_name in SERVICES_COLLECT_LIST
+        ]
+        for service_name in service_names:
+            builder.submit_work(GcpSku.collect_resources, builder, parent=service_name)
+
+        return result
+
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        for sku in builder.resources_of(GcpSku):
```

Comment:
`resources_of` is expensive. Consider passing a function to `add_edge` and `node` which can perform the filtering.